src/platform/plugins/shared/data_views/server/rest_api_routes/internal/fields_for.ts

```diff
@@ -20,6 +21,19 @@ import type {
 import type { FieldDescriptorRestResponse } from '../route_types';
 import { FIELDS_FOR_WILDCARD_PATH as path } from '../../../common/constants';
 
+/**
+ * Copied from `@kbn/data-plugin/server` to avoid a cyclic dependency.
+ *
+ * A simple utility function that returns an `AbortSignal` corresponding to an `AbortController`
+ * which aborts when the given request is aborted.
+ * @param aborted$ The observable of abort events (usually `request.events.aborted$`)
+ */
+function getRequestAbortedSignal(aborted$: Observable<void>): AbortSignal {
+  const controller = new AbortController();
+  aborted$.subscribe(() => controller.abort());
+  return controller.signal;
+}
```

Comment:
Variants of this function are spread around the Kibana code base in some places. I copied this to limit the scope of this PR, but it might be worth moving this function to a package and clean up across Kibana in a follow up.

Reply:
Agreed, good opportunity for cleanup, but can be handled separately.